examples/pytorch/graphsage/advanced/train_sampling_unsupervised.py

```diff
@@ -133,6 +133,7 @@ def run(proc_id, n_gpus, args, devices, data):
         # blocks.
         tic_step = time.time()
         for step, (input_nodes, pos_graph, neg_graph, blocks) in enumerate(dataloader):
+            input_nodes = input_nodes.to(nfeat.device)
```

Comment:
When features are on the CPU, `input_nodes` will first be copied to GPU in the dataloader and then copied back to the CPU for indexing. It's not caused by this PR and can be eliminated by unifying `--graph_device` and `--data_device` just like other examples. Perhaps we should take a note and fix it when we refactor this example in the future.

Reply:
I generally agree. It does sound odds that copying `input_nodes` back and forth...  But I think it cannot be avoided by specifying graph/data devices. The `input_nodes` always belongs to the same `device` as dataloader and model (`--gpu`), which is not configured in the same way as nfeat (`--data-device`) and g (`--graph-device`).

Reply:
With the new sampling pipeline, we can specify `prefetch_node_features` to the sampler and don't need `input_nodes = input_nodes.to(nfeat.device)` anymore.

Reply:
I see... Thanks Xin for noting this. I will keep it in mind :)